elements/rh-alert/rh-alert.ts

```diff
@@ -117,10 +117,11 @@ export class RhAlert extends LitElement {
     const hasActions = this.#slots.hasSlotted('actions');
     const hasBody = this.#slots.hasSlotted(SlotController.anonymous as unknown as string);
     return html`
-      <div id="container"
+      <rh-surface id="container"
```

Comment:
Instead of using a `rh-surface` element in the shadow root I think we'd want to make `rh-alert` itself a provider/consumer.  By doing so it essentially becomes a surface and can pass context down to its children like the buttons that are slotted.   

```
import { colorContextProvider, type ColorPalette } from '../../lib/context/color/provider.js';
import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
...

@colorContextProvider()
@property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette = 'lightest';

@colorContextConsumer() private on?: ColorTheme;
```

The text can managed by just adding:

```
#container {
  color: var(--rh-color-text-primary-on-light, #151515) !important;
}
```

That said I was trying to implement with hope it would work without reflection of the `color-palette` attribute, essentially making that context invisible to the user.  That said as soon as `reflect: false` is set the slotted buttons no longer receive the correct context.  /cc @bennypowers thoughts?

Reply:
let's do this some other time